services/claim-relayer/scripts/mainSubscriber.ts

```diff
@@ -0,0 +1,140 @@
+import type { Api } from "@cennznet/api";
+import type { BaseProvider } from "@ethersproject/providers";
+
+import mongoose from "mongoose";
+import amqp from "amqplib";
+import { Keyring } from "@polkadot/keyring";
+import {
+	CENNZNET_NETWORK,
+	CENNZNET_SIGNER,
+	MONGODB_SERVER,
+	RABBBITMQ_SERVER,
+	TOPIC_CENNZnet_CONFIRM,
+	TOPIC_VERIFY_CONFIRM,
+} from "@/libs/constants";
+import { getLogger, sendSlackAlert } from "@/libs/utils";
+import {
+	retryMessage,
+	sendCENNZnetClaimSubscriber,
+	subscribeFinalizedBlock,
+	verifyClaimSubscriber,
+} from "@claim-relayer/utils";
+
+const logger = getLogger("ClaimSubscriber");
+
+let nonce: number;
+let firstMessage = true;
+
+export async function mainSubscriber(
+	cennzApi: Api,
+	ethersProvider: BaseProvider
+): Promise<void> {
+	if (mongoose.connection.readyState !== 1)
+		await mongoose.connect(MONGODB_SERVER);
+	const rabbit = await amqp.connect(RABBBITMQ_SERVER);
+	logger.info(
+		`Rabbit MQ Connected to Host:  ${(rabbit.connection as any).stream._host}`
+	);
+	logger.info(`Connected to cennznet network ${CENNZNET_NETWORK}`);
+	const signer = new Keyring({ type: "sr25519" }).addFromSeed(
+		CENNZNET_SIGNER as any
+	);
+	nonce = (
+		await cennzApi.rpc.system.accountNextIndex(signer.address)
+	).toNumber();
+
+	// Keep track of latest finalized block
+	subscribeFinalizedBlock(cennzApi, logger);
+
+	//Setup rabbitMQ
+	const consumerMessageLimit = 10;
+	const messageTimeout = 60000 * 5; //5 minutes
```

Comment:
might be worth adding these to constants file as well and making optional env vars for them